src/Compilers/CSharp/Test/Semantic/Semantics/PatternMatchingTests5.cs

```diff
@@ -0,0 +1,333 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.CodeAnalysis.CSharp.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.CSharp.UnitTests
+{
+    public class PatternMatchingTests5 : PatternMatchingTestBase
+    {
+        [Fact]
+        public void ExtendedPropertyPatterns_01()
+        {
+            var program = @"
+using System;
+class C
+{
+    public C Prop1 { get; set; }
+    public C Prop2 { get; set; }
+    public C Prop3 { get; set; }
+    
+    static bool Test1(C o) => o is { Prop1.Prop2.Prop3: null };
+    static bool Test2(S o) => o is { Prop1.Prop2.Prop3: null };
+    static bool Test3(S? o) => o is { Prop1.Prop2.Prop3: null };
+    static bool Test4(S0 o) => o is { Prop1.Prop2.Prop3: 420 };
```

Comment:
>Test4 [](http://example.com/codeflow?start=16&length=5)

Consider adding `VerifyIL("C.Test4")`. I think it would be interesting to see the difference between that generated code and `Test2`.